rfcs/apollo-federation-error-codes.md

```diff
@@ -0,0 +1,136 @@
+| 🔴     | INVALID_LINK_DIRECTIVE_USAGE      | An application of the @link directive is invalid/does not respect the specification.                                                                                                                                                                                                                                                                                                                                                                                                                                       |
+| 🔴     | INVALID_LINK_IDENTIFIER           | A URL/version for a @link feature is invalid/does not respect the specification.                                                                                                                                                                                                                                                                                                                                                                                                                                           |
+| 🔴     | INVALID_SUBGRAPH_NAME             | A subgraph name is invalid. (Subgraph names cannot be a single underscore (\*)).                                                                                                                                                                                                                                                                                                                                                                                                                                           |
+| 🔴     | LINK_IMPORT_NAME_MISMATCH         | The import name for a merged directive (as declared by the relevant @link(import:) argument) is inconsistent between subgraphs.                                                                                                                                                                                                                                                                                                                                                                                            |
+| 🔴     | REQUIRES_FIELDS_MISSING_EXTERNAL  | The fields argument of a @requires directive includes a field that is not marked as @external.                                                                                                                                                                                                                                                                                                                                                                                                                             |
```

Comment:
```graphql
extend schema
              @link(
                url: "https://specs.apollo.dev/federation/v2.3"
                import: ["@key", "@requires"]
              )

            type Query {
              users: [User]
            }

            type User @key(fields: "id") {
              id: ID!
              internalId: ID!
              profile: Profile @requires(fields: "internalId")
            }

            type Profile {
              name: String
            }
```

Reply:
```
[users] On field "User.profile", for @requires(fields: "internalId"): field "User.internalId" should not be part of a @requires since it is already provided by this subgraph (it is not marked @external)
```